services/jobs/src/main/java/com/linkedin/openhouse/jobs/api/validator/impl/OpenHouseJobsApiValidator.java

```diff
@@ -51,19 +51,19 @@ public void validateCreateJob(CreateJobRequestBody createJobRequestBody) {
 
   // inner class to validate jobConf fields.
   private static class JobConfValidator {
-    private static boolean validateExecutionConfig(Map<String, String> conf) {
-      String memory = conf.getOrDefault(JOB_MEMORY_CONFIG, "");
-      return memory.matches(SPARK_MEMORY_REGEX_ALLOW);
+    private static boolean validateSparkKeys(Map<String, String> conf) {
+      for (Map.Entry<String, String> entry : conf.entrySet()) {
+        if (!entry.getKey().startsWith("spark.")) {
```

Comment:
A few questions on the purpose of only validating the keys to have `spark.` prefix:
1. Failing the validation with keys without `spark.` seems restrictive and a strict check which will fail jobs API call. Why not just ignore anything which is not `prefixed` with `spark.` since tableProperties are not restricted in tablesApi client side and user can add other things too.
2. Should the values also have some form of validation to avoid failing the job coz of that?
3. With this, we are opening spark run properties to end users, should the list of properties that can be added be more deterministic?

Reply:
Since I have been tuning Spark for various job types, I think the value of flexible configuration is very high relative to the value of having stricter validation, because I don't think we have a good idea what the validation that is not too restrictive looks like. We can make it more restrictive in the future if it's valuable.

Reply:
Can we consider not failing the job if maintenance properties are not prefixed with `spark.`? Rather, filter and pick.

Reply:
After giving it another thought, I think we can start w/o any validation and filtering. I can see that we might want to be able to tune Iceberg properties as well when rolling out MoR

Reply:
cc @cbb330 @autumnust